mlrun/model_monitoring/applications/histogram_data_drift.py

```diff
@@ -146,31 +147,41 @@ def _compute_metrics_per_feature(
 
         return metrics_per_feature
 
-    def _add_general_drift_result(
+    def _get_general_drift_result(
         self,
         metrics: list[mm_results.ModelMonitoringApplicationMetric],
-    ) -> None:
-        """Add the general drift result to the results list and log it"""
-        value = np.mean(
-            [
-                metric.value
-                for metric in metrics
-                if metric.name
-                in [
-                    f"{HellingerDistance.NAME}_mean",
-                    f"{TotalVarianceDistance.NAME}_mean",
+        monitoring_context: mm_context.MonitoringApplicationContext,
+        metrics_per_feature: DataFrame,
+    ) -> mm_results.ModelMonitoringApplicationResult:
+        """Get the general drift result from the metrics list"""
+        value = cast(
+            float,
+            np.mean(
+                [
+                    metric.value
+                    for metric in metrics
+                    if metric.name
+                    in [
+                        f"{HellingerDistance.NAME}_mean",
+                        f"{TotalVarianceDistance.NAME}_mean",
+                    ]
                 ]
-            ]
+            ),
         )
 
         status = self._value_classifier.value_to_status(value)
-        metrics.append(
-            mm_results.ModelMonitoringApplicationResult(
-                name=HistogramDataDriftApplicationConstants.GENERAL_RESULT_NAME,
-                value=value,
-                kind=ResultKindApp.data_drift,
-                status=status,
-            )
+        return mm_results.ModelMonitoringApplicationResult(
+            name=HistogramDataDriftApplicationConstants.GENERAL_RESULT_NAME,
+            value=value,
+            kind=ResultKindApp.data_drift,
+            status=status,
+            extra_data={
+                EventFieldType.CURRENT_STATS: json.dumps(
+                    monitoring_context.feature_stats
+                ),
+                EventFieldType.DRIFT_MEASURES: metrics_per_feature.T.to_json(),
+                EventFieldType.DRIFT_STATUS: status.value,
+            },
         )
 
     def _get_metrics(
```

Comment:
just a recommendation you can convert both _get_metrics & _log_json_artifact into staticmethods.
```suggestion
    @staticmethod
    def _get_metrics(
```

Reply:
d720750